geomstats/geometry/special_euclidean.py

```diff
@@ -115,12 +122,32 @@ def belongs(self, point, point_type=None):
             point = gs.to_ndarray(point, to_ndim=2)
             n_points, point_dim = point.shape
             belongs = point_dim == self.dimension
-            belongs = gs.to_ndarray(belongs, to_ndim=1)
-            belongs = gs.to_ndarray(belongs, to_ndim=2, axis=1)
-            belongs = gs.tile(belongs, (n_points, 1))
+            belongs = gs.logical_and(belongs,
+                                     self.rotations.belongs(point[:, :self.n]))
+            # TODO: Should to_ndarray do this if to_ndim = ndim(x) - 1 ?
+            belongs = gs.flatten(belongs)
         elif point_type == 'matrix':
             point = gs.to_ndarray(point, to_ndim=3)
-            raise NotImplementedError()
+            n_points, point_dim1, point_dim2 = point.shape
+            # Check dimensions
```

Comment:
No comments! Here are the reasons:
- Different developers have very different comment styles and the codebase can become a giant mess if everybody comments,
- Comments are difficult to maintain, they are not unit-tested for example, and you might have comments remaining that have nothing to do with the next code line in a few weeks from now,
- Prohibiting comments forces the collaborators to have a cleaner code! If you feel that your code is not understandable, maybe: find better variable names? add an auxiliary function, with a good function name, that has a docstring that explains what is going on? etc.

This no-comment guideline is a guideline used in software engineering at Google and many other startups that I know of in SF.

Reply:
Challenge accepted !

Is this rule also true with "uglier" languages (say, C++) ?

Reply:
Good question... I don't actually know, but I could very well be different!